autonomous_software_pkg/src/target_generator.py

```diff
@@ -135,17 +141,37 @@ def callback_current_pose(self, pose_msg: PoseStamped):
             """ Publish target curvature """
             self.publish_target_curvature(curvature)
 
+            """ Compute target speed """
+            if self.velocity_mode == "MANUAL_INPUT_SPEED":
+                target_speed = self.user_input_speed_mps
+            elif self.velocity_mode == "WAYPOINT_FOLLOWING":
+                try:
+                    if self.id_closest_wp == self.waypoints.waypoints[-1].id:
+                        target_speed = 0
+                    else:
+                        target_speed = (
+                            self.speed_gain
+                            * self.waypoints.waypoints[self.id_closest_wp].speed_mps
+                        )
+                except IndexError:
+                    rospy.logwarn(
+                        "An error occured while retrieving speed from waypoint"
+                    )
+                    target_speed = 0
+
             """ Publish target speed """
-            # Note: Strategy to change. Will likely not use the targetPoint
-            self.publish_target_speed(targetPoint)
+            self.publish_target_speed(target_speed)
 
     def callback_current_velocity(self, twist_msg: TwistStamped):
         """ """
         pass
 
     def callback_manual_input_speed(self, twist_msg: TwistStamped):
```

Comment:
Why is it a TwistStamped and not just a float btw?

Reply:
I don't think there was a particular reason when we set this in the diagram. Maybe we did it for similitude with the /current_velocity topic. 
I think having an information about the time the topic is published could be useful (for example if we want to discard messages received too long ago), so I guess it could be a FloatStamped (custom message) or TwistStamped.